services/checks/checks.go

```diff
@@ -214,7 +214,12 @@ func (s *Service) resendAlerts(ctx context.Context) {
 	defer t.Stop()
 
 	for {
-		s.alertmanagerService.SendAlerts(ctx, s.alertsRegistry.collect())
+		settings, err := models.GetSettings(s.db)
+
+		// Only sends if IAs enabled in the settings
+		if (err == nil) && settings.IntegratedAlerting.Enabled {
+			s.alertmanagerService.SendAlerts(ctx, s.alertsRegistry.collect())
+		}
```

Comment:
`alertsRegistry` contains only STT alerts, IA alerts sent to `alertmanager` directly from `vmalert`. Moreover now disabling IA will also disables STT, but it's two independent features, except they both use alertmanager. 

Tickets original text proposes more o less right solution. All generated files should be removed (but what we will do with user-created files?) and vmalert should be asked to reload the configuration.

Reply:
Reworked. Please check.